ADAL/src/ADDefaultEvent.m

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "ADDefaultEvent.h"
+#import "ADEventInterface.h"
+
+@implementation ADDefaultEvent
+
+@synthesize propertyMap = _propertyMap;
+
+- (id)init
+{
+    //Ensure that the appropriate init function is called. This will cause the runtime to throw.
+    [super doesNotRecognizeSelector:_cmd];
+    return nil;
+}
+
+- (id)initWithName:(NSString*)eventName
+{
+    if (!(self = [super init]))
+    {
+        return nil;
+    }
+    
+    _propertyMap = [[self defaultParameters] mutableCopy];
+    [_propertyMap addObject:@[@"event_name", eventName]];
+    
+    return self;
+}
+
+- (void)setProperty:(NSString*)name value:(NSString*)value
+{
+    if ([NSString adIsStringNilOrBlank:name] || [NSString adIsStringNilOrBlank:value])
+    {
+        return;
+    }
+    
+    [_propertyMap addObject:@[name, value]];
+}
+
+- (NSArray*)getProperties
+{
+    return _propertyMap;
+}
+
+- (void)setStartTime:(NSDate*)time
```

Comment:
Getting the system current time can be done in either  Event or Telemetry. To make it consistent across platform, I can move it to class Telemetry if we all choose it that way :)